core/src/indexer/fetch_logs.rs

```diff
@@ -36,99 +39,249 @@ pub fn fetch_logs_stream(
-                    }
+                        }
 
-                    if let Some(result) = result {
-                        current_filter = result.next;
-                        max_block_range_limitation = result.max_block_range_limitation;
-                    } else {
-                        break;
+                        if let Some(result) = result {
+                            current_filter = result.next;
+                            max_block_range_limitation = result.max_block_range_limitation;
+                        } else {
+                            break;
+                        }
+                    }
+                    Err(e) => {
+                        error!(
+                            "{} - {} - Semaphore error: {}",
+                            &config.info_log_name,
+                            IndexingEventProgressStatus::Syncing.log(),
+                            e
+                        );
+                        continue;
                     }
-                }
-                Err(e) => {
-                    error!(
-                        "{} - {} - Semaphore error: {}",
-                        &config.info_log_name,
-                        IndexingEventProgressStatus::Syncing.log(),
-                        e
-                    );
-                    continue;
                 }
             }
-        }
 
-        info!(
-            "{} - {} - Finished indexing historic events",
-            &config.info_log_name,
-            IndexingEventProgressStatus::Completed.log()
-        );
-
-        // Live indexing mode
-        if config.live_indexing && !force_no_live_indexing {
-            live_indexing_stream(
-                &config.network_contract.cached_provider,
-                &tx,
-                &contract_address,
-                &config.topic_id,
-                &config.indexing_distance_from_head,
-                current_filter,
+            info!(
+                "{} - {} - Finished indexing historic events",
                 &config.info_log_name,
-                &config.semaphore,
-                config.network_contract.disable_logs_bloom_checks,
-            )
-            .await;
+                IndexingEventProgressStatus::Completed.log()
+            );
+
+            // Live indexing mode
+            if config.live_indexing && !force_no_live_indexing {
+                live_indexing_stream(
+                    &config.network_contract.cached_provider,
+                    &tx,
+                    &contract_address,
+                    &config.topic_id,
+                    &config.indexing_distance_from_head,
+                    current_filter,
+                    &config.info_log_name,
+                    &config.semaphore,
+                    config.network_contract.disable_logs_bloom_checks,
+                )
+                .await;
+            }
         }
     });
 
     UnboundedReceiverStream::new(rx)
 }
 
+async fn process_exex_stream(
```

Comment:
maybe worth spinning out an reth.rs file to hold these things in may be easier to find it etc?